### ISSUE TITLE
10x notepad perforce integration does not work well

### ISSUE THREAD

**Issue by MarcAudy** [slynch8/10x#1586]
When using 10x notepad there is no solution root folder and so P4CLIENT is not used.
Because you can be interacting with many different files from different streams, the desired behavior is that the p4 settings would be pulled from the P4CONFIG for whichever file you are interacting with at that time. It seems like 10x notepad should be either connecting to perforce on a per file basis or simply on demand for actions with any given file, using the p4 settings as dictated by the P4CONFIG for that file path.

**Comment by slynch8** [comment]
Fixed (1.0.4). I've made it always check for the P4CLIENT file on OpenForEdit. One thing to keep in mind is that it doesn't reset the connection back, so it will stay on the last connection. Let me know if this caused any problems.

**Comment by MarcAudy** [comment]
Unfortunately it doesn't seem to work as I would hope it does.
The circumstance I'd like to see work is when I have files in 2 different folders that are in different streams and so the P4CONFIG values for them are different. Ideally all the perforce operations for the file would just work based on the file path.
With the most recent changes it seems like it can correctly deal with the first file I interact with, but then when I go to interact with the other file the perforce operations cannot succeed.

**Comment by slynch8** [comment]
It should be checking for the P4CONFIG on every file that you check out now. In the latest build (1.0.5) I've made it log out to the log file the p4config file that it's using, can you check to see if it's finding the right file. In the .log file in %appdata%\10x\ look for "Found P4CONFIG file"

**Comment by MarcAudy** [comment]
Yup, my mistake. I managed to pick a file from one workspace that I didn't
have p4.ini properly set up for and so it was falling back to a different
ini that was unexpected.
A few more experiments and it does actually seem to be working as I'd hoped.

**Comment by slynch8** [comment]
That's a good point, I'll need to add the p4config check for any file operation, I only did it for checkout. Ok, I'll add that on Monday. Oh dear, big fail on the error message box! I'll take a look :-)

**Comment by slynch8** [comment]
fixed (1.0.10) all commands now check the p4config file
